### DISCUSSION TITLE
I can't fetch a user's permsissions from a server

### DISCUSSION THREAD

**Discussion by Scarner56** [discordjs/discord.js#6534] [Q&A] [answered]
**Please describe the problem you are having in as much detail as possible:**
Trying to fetch a user's permissions from a server
**Include a reproducible code sample here, if possible:**

```js
app.get("/yonet", async(req, res) => {
  if(!req.user) return res.redirect("/hata?type=giris")
  let id = req.query.server
  if(!id) return res.redirect("/hata?type=bulunamadı")
  let p =  new Promise(async (resolve, reject) => {
    let sunucu = await client.guilds.cache.get(id)
    let a = await sunucu.members.cache.get(req.user.id)
    console.log(a)
    if(a.permissions.has("MANAGE_GUILD")){
      resolve()
    }else {
      reject()
    }
    p.then((pe) => {
      console.log(pe)
    })
  }) 
    renderTemplate(res, req, "yonet/server.ejs", {id})
```
**The errorr**

```js
    if(a.permissions.has("MANAGE_GUILD")){
         ^

TypeError: Cannot read property 'permissions' of undefined
```

**Further details:**

- discord.js version: 13.1.0
- Node.js version: 16+
- Operating system: ubuntu 18
- Priority this issue should have – please be realistic and elaborate if possible:

**Relevant client options:**

- partials: none
- gateway intents: server members
- other: none

**Comment by Scarner56** [comment]
Btw i asked this error on discord and no one else found solution

**Comment by Jiralite** [comment]
This doesn't seem like a bug, but a lack of understanding. Feel free to make a new [discussion](https://github.com/discordjs/discord.js/discussions/new) about it, as this issue tracker is for bug reports and enhancement suggestions.

**Comment by Scarner56** [comment]
Should i close this issue?

**Comment by kyranet** [comment]
Don't worry, I can just do this.

**Comment by Jiralite** [comment, marked as answer]
Okay, there are a lot of problems here.

1. The provided reproducible code sample doesn't work. It should be something that we can copy & paste to test ourselves to see the error. It's also incorrect syntax.
2. `p` is defined as a `Promise`, but the code in there doesn't need it to be. You also include a reference to itself in there.
3. You have unnecessary `await`s here:
```JavaScript
let sunucu = await client.guilds.cache.get(id)
let a = await sunucu.members.cache.get(req.user.id)
```
These methods don't return a `Promise` (which adds to point 2).

4. You specify the "server members" intent which doesn't exist - I assume you mean the `GUILD_MEMBERS` one. I feel like you should be including `GUILDS` too.

After all of this, I assume the problem is that your member is not cached. Code needs tidying!
